plugin/auto_mkdir2.vim

```diff
@@ -24,11 +27,14 @@ endif
 "##########################################################
 " Commands
 command! -nargs=* -complete=dir MkdirP call s:mkdir_p(<q-args>, 1)
+command! -nargs=* -complete=dir MkdirPC call s:mkdir_p(<q-args>, v:cmdbang)
 
-augroup auto_mkdir2
-	autocmd!
-	autocmd BufWritePre * call s:mkdir_p(expand("<amatch>:p:h"), v:cmdbang)
-augroup end
+if g:auto_mkdir2_allfiles == "1"
```

Comment:
`g:auto_mkdir2_noautocmd` is probably a better name for this, I think.

It's also better to just use `if g:auto_mkdir2_allfiles` instead of the comparison to the string `"1"`; that way setting it to `v:true` or `"true"` or whatnot will also work.

Reply:
Actually, `g:auto_mkdir2_autocmd` with a default of `1` is even better, as settings with `no` in them look kinda weird IMO.

You don't really need to define a default at the top cluttering the global namespace with settings people never touched, using `get()` with a default is usually the best way:

    if get(g:, `auto_mkdir2_autocmd`, 1)
       augroup [...]
    endif

and then people can add `let g:auto_mkdir2_autocmd = 0` in their plugins to disable it.

This is an old-ish plugin I haven't touched in a while and predates from when I figured out there's a better way to do this 🙃

Reply:
Or maybe the default should be `*`, and then you can do `let g:auto_mkdir2_autocmd = '*.md'` to easily apply it to Markdown files only, and an empty string would be to disable it entirely 🤔

Maybe I should think more before writing comments haha

Reply:
Changing the flag to `g:auto_mkdir2_autocmd` and not cluttering the namespace make a lot of sense so I implemented that.

I couldn't find a way to set the `autocmd `pattern from a variable without using `execute` which is probably not what you want.

What is the better way you found to do this?

Reply:
Did you push your changes? I'm not seeing them?

Reply:
I didn't push changes. I wanted to resolve all your concerns before I pushed anything